languages/ruby/lib/oso/polar/query.rb

```diff
@@ -97,13 +70,33 @@ def application_error(message)
       # @param call_id [Integer]
       # @param instance [Hash<String, Object>]
       # @raise [Error] if the FFI call raises one.
-      def handle_call(attribute, call_id:, instance:, args:, kwargs:)
-        register_call(attribute, call_id: call_id, instance: instance, args: args, kwargs: kwargs)
-        result = JSON.dump(next_call_result(call_id))
+      def handle_call(attribute, call_id:, instance:, args:, kwargs:) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+        instance = host.to_ruby(instance)
+        args = args.map { |a| host.to_ruby(a) }
+        kwargs = Hash[kwargs.map { |k, v| [k.to_sym, host.to_ruby(v)] }]
+        # The kwargs.empty? check is for Ruby < 2.7.
+        result = if kwargs.empty?
+                   instance.__send__(attribute, *args)
+                 else
+                   instance.__send__(attribute, *args, **kwargs)
+                 end
+        result = JSON.dump(host.to_polar(result))
         call_result(result, call_id: call_id)
-      rescue InvalidCallError => e
+      rescue ArgumentError, NoMethodError => e
         application_error(e.message)
         call_result(nil, call_id: call_id)
+      end
+
+      def handle_next_external(call_id, iterable)
+        unless calls.key? call_id
+          value = host.to_ruby iterable
+          raise InvalidIteratorError unless value.is_a? Enumerable
```

Comment:
Is `Enumerable` really the only check here? Not familiar enough with Ruby.

Reply:
As far as I can tell, that's the main trait. @gj ?

Reply:
Cursory google search says that's correct

Reply:
I think this is fine -- the call to `.lazy` on 95 will turn any `Enumerable` into a lazy enumerator. So this check is basically asking "is it possible to create an enumerator out of `value`, even if it isn't already an enumerator?"